### ISSUE TITLE
Cursor moved to end after any change

### ISSUE THREAD

**Issue by conraddamon** [lovasoa/react-contenteditable#191]
1. Type a sentence such as `Cursor to end after any change`
2. Realize you forgot to type the word `moved`
3. Click to position the cursor in front of the word `to`
4. Type `m`.

After the character is added to the input's content, the cursor is moved to the end and you have to click to place it back after the letter you just typed (and any other letters you want to insert).

**Comment by lovasoa** [comment]
Hello,
  Sorry, but I cannot reproduce this error. Please try it on https://codesandbox.io/s/4rlw34mnk7
  If you cannot reproduce, then probably it's your code that it changing the innerHTML while the user is typing, causing the caret jump.

**Comment by conraddamon** [comment]
That's true, but I figured this would be a common use case. We take the text the user has typed and convert it to highlighted markup (using colored spans), so we are changing the HTML in the component any time the input changes. That results in a call to `replaceCaret`, which always moves the cursor to the end. Sometimes, that's not desirable.

The way I have handled this (in my own content-editable component, separate from this library) is to read the cursor position before making the update, then restore it afterward.
